Can't build docker container image

After migration to distroless, the container image is not building properly. Dockerfile uses RUN command to add a metrics-server user, but there is no underlying shell that could run this command:

```
Step 1/5 : FROM gcr.io/distroless/static:latest
latest: Pulling from distroless/static
Digest: sha256:9b60270ec0991bc4f14bda475e8cae75594d8197d0ae58576ace84694aa75d7a
Status: Image is up to date for gcr.io/distroless/static:latest
 ---> dbd9343fe273
Step 2/5 : COPY metrics-server /
 ---> 0b3e7b19689d
Step 3/5 : RUN adduser -D metrics-server
 ---> Running in cd29f0bc3ad6
OCI runtime create failed: container_linux.go:344: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory": unknown
make: *** [Makefile:56: container-amd64] Error 1
```

/assign @yuwenma 
Yuwen, can you take a look?

--- a/deploy/docker/Dockerfile
+++ b/deploy/docker/Dockerfile
@@ -2,8 +2,4 @@ FROM BASEIMAGE
 
 COPY metrics-server /
 
-RUN adduser -D metrics-server
-
-USER metrics-server
-
 ENTRYPOINT ["/metrics-server"]

